scio-extra/src/main/scala/com/spotify/scio/extra/csv/CsvIO.scala

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2019 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.scio.extra.csv
+
+import java.io.{Reader, Writer}
+import java.nio.channels.{Channels, WritableByteChannel}
+import java.nio.charset.StandardCharsets
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.coders.Coder
+import com.spotify.scio.io._
+import com.spotify.scio.util.ScioUtil
+import com.spotify.scio.values.SCollection
+import kantan.csv.CsvConfiguration.{Header, QuotePolicy}
+import kantan.csv._
+import kantan.csv.engine.ReaderEngine
+import org.apache.beam.sdk.io.FileIO.ReadMatches.DirectoryTreatment
+import org.apache.beam.sdk.io.FileIO.ReadableFile
+import org.apache.beam.sdk.io.fs.{MatchResult, MetadataCoderV2}
+import org.apache.beam.sdk.io.{FileIO, ReadableFileCoder}
+import org.apache.beam.sdk.transforms.{DoFn, ParDo}
+import org.apache.beam.sdk.transforms.DoFn.ProcessElement
+import org.apache.beam.sdk.{io => beam}
+
+/**
+ * This package uses a CSV mapper called [[https://nrinaudo.github.io/kantan.csv/ Kantan]].
+ *
+ * import the following:
+ * {{{
+ *   import kantan.csv._
+ *   import com.spotify.scio.extra.csv._
+ * }}}
+ *
+ * =Reading=
+ * == with a header ==
+ * [[https://nrinaudo.github.io/kantan.csv/headers.html Kantan docs]]
+ * {{{
+ *  case class User(name: String, age: Int)
+ *  implicit val decoder = HeaderDecoder.decoder("fullName", "userAge")(User.apply _)
+ *  val users: SCollection[User] = scioContext.csvFile(path)
+ * }}}
+ *
+ * == without a header ==
+ * [[https://nrinaudo.github.io/kantan.csv/rows_as_arbitrary_types.html Kantan docs]]
+ * {{{
+ *  case class User(name: String, age: Int)
+ *  implicit val decoder = HeaderDecoder.ordered { (name: String, age: Int) => User(name, age) }
+ *  val users: SCollection[User] = scioContext.csvFile(path)
+ * }}}
+ *
+ * =Writing=
+ * == with a header ==
+ * [[https://nrinaudo.github.io/kantan.csv/headers.html Kantan docs]]
+ * {{{
+ *   import kantan.csv._
+ *   import com.spotify.scio.extra.csv._
+ *   implicit val encoder = HeaderEncoder.caseEncoder("fullName", "age")(User.unapply)
+ *
+ *   val users: SCollection[User] = ???
+ *   users.saveAsCsvFile(path)
+ * }}}
+ * == without a header ==
+ * [[https://nrinaudo.github.io/kantan.csv/headers.html Kantan docs]]
+ * {{{
+ *   import kantan.csv._
+ *   import com.spotify.scio.extra.csv._
+ *   implicit val encoder = HeaderEncoder.caseEncoder("fullName", "age")(User.unapply)
```

Comment:
It looks like this example is the same as the example for the `with a header` section?